Lib/fontTools/ttLib/ttGlyphSet.py

```diff
@@ -1,158 +1,193 @@
 
-        Args:
-                        font (TTFont): The font object (used to get metrics).
-                        glyphs (dict): A dictionary mapping glyph names to ``_TTGlyph`` objects.
-                        glyphType (class): Either ``_TTGlyphCFF`` or ``_TTGlyphGlyf``.
-        """
-        self._glyphs = glyphs
-        self._hmtx = ttFont["hmtx"]
-        self._vmtx = ttFont["vmtx"] if "vmtx" in ttFont else None
-        self._glyphType = glyphType
+            self.hvarTable = getattr(font.get("HVAR"), "table", None)
+            if self.hvarTable is not None:
+                self.hvarInstancer = VarStoreInstancer(
+                    self.hvarTable.VarStore, font["fvar"].axes, location
+                )
+            # TODO VVAR, VORG
+
+    def __contains__(self, glyphName):
+        return glyphName in self.glyphsMapping
 
-    def keys(self):
-        return list(self._glyphs.keys())
+    def __iter__(self):
+        return iter(self.glyphsMapping.keys())
 
+    def __len__(self):
+        return len(self.glyphsMapping)
+
+    @deprecateFunction(
+        "use 'glyphName in glyphSet' instead", category=DeprecationWarning
+    )
     def has_key(self, glyphName):
-        return glyphName in self._glyphs
+        return glyphName in self.glyphsMapping
 
-    __contains__ = has_key
+
+class _TTGlyphSetGlyf(_TTGlyphSet):
+    def __init__(self, font, location):
+        self.glyfTable = font["glyf"]
+        super().__init__(font, location, self.glyfTable)
+        if location:
+            self.gvarTable = font.get("gvar")
 
     def __getitem__(self, glyphName):
-        horizontalMetrics = self._hmtx[glyphName]
-        verticalMetrics = self._vmtx[glyphName] if self._vmtx else None
-        return self._glyphType(
-            self, self._glyphs[glyphName], horizontalMetrics, verticalMetrics
-        )
+        return _TTGlyphGlyf(self, glyphName)
 
-    def __len__(self):
-        return len(self._glyphs)
 
-    def get(self, glyphName, default=None):
-        try:
-            return self[glyphName]
-        except KeyError:
-            return default
+class _TTGlyphSetCFF(_TTGlyphSet):
+    def __init__(self, font, location):
+        tableTag = "CFF2" if "CFF2" in font else "CFF "
+        self.charStrings = list(font[tableTag].cff.values())[0].CharStrings
+        super().__init__(font, location, self.charStrings)
+        self.blender = None
+        if location:
+            from fontTools.varLib.varStore import VarStoreInstancer
+
+            varStore = getattr(self.charStrings, "varStore", None)
+            if varStore is not None:
+                instancer = VarStoreInstancer(
+                    varStore.otVarStore, font["fvar"].axes, location
+                )
+                self.blender = instancer.interpolateFromDeltas
+
+    def __getitem__(self, glyphName):
+        return _TTGlyphCFF(self, glyphName)
 
 
-class _TTGlyph(object):
+class _TTGlyph:
```

Comment:
maybe we could make _TTGlyph an abstract base class and define draw/drawPoints as abstract methods that must be overridden by subclasses

Reply:
FWIW, also `_TTGlyphSet` in theory is abstract, with particular subclasses needing to override `__getitem__` at least

Reply:
Ah yes, makes sense for `__TTGlyph`, will do.

`_TTGlyphSet` is already an actual abstract baseclass, via `collections.abc.Mapping`.